include/eeprom/core/data_rev.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include "accessor.hpp"
+#include "addresses.hpp"
+#include "common/core/bit_utils.hpp"
+#include "task.hpp"
+
+namespace eeprom {
+namespace data_revision {
+
+using DataRevisionType = std::vector<uint8_t>;
```

Comment:
so we definitely can't use vectors that call `malloc` since we want to avoid that in a microcontroller context. we do have a memory pool allocator implementation, but those work best for, well, memory pool contexts - situations where we have a bunch of objects of the same size and repeatedly want to fill them. 

i think that the needs of the interface can be better fulfilled by using iterator pairs or standard ranges. callers can then create arrays of the right size for the specific data they're trying to retrieve or write since they know what the data is, while the interface itself uses the iterpairs/ranges to preserve generality.